### ISSUE TITLE
Parsing

### ISSUE THREAD

**Issue by kengdoj** [SSAgov/ANDI#20]
Will ANDI test for the Parsing requirement (WCAG Level A SC 4.1.1)?

There is a [Parsing bookmarklet](https://developer.paciellogroup.com/blog/2012/02/wcag-2-0-parsing-error-bookmarklet/) available (originally developed in 2012). I'm not aware of any plans to update the bookmarklet.

Additional info is on this [Parsing Criterion](https://developer.paciellogroup.com/blog/2015/11/wcag-2-0-parsing-criterion-is-a-pita/) page.

**Comment by SSAgov** [comment]
The short answer is no.
 
Regarding missing end tags: if an end tag is forgotten, a browser will automatically assume where an end tag should be. Checking for start and end tag needs to be done at the source level, not at the parsed code level in the DOM - which is what ANDI analyzes. 
 
With duplicate attributes, browsers today simply reject (ignore) a duplicate attribute. Again, this would have to be done at the source level, since at the time of the DOM inspection, any duplicate attributes are not present.
 
Any type of HTML validator (https://validator.w3.org/) should do the tasks above.
 
ANDI does do some nesting checking only when it suspects an accessibility issue. ANDI also will check for duplicate IDs - again only when it relates to a potential accessibility issue.
